File: src/backend/Domain/Models/Actor.cs
namespace Domain.Models;

public class Actor
{
    public Guid Id { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Biography { get; set; }
    public DateOnly BirthDate { get; set; }
    public string? PhotoUrl { get; set; }
    public List<Photo>? Photos { get; set; } = new();
    public List<MovieActor> Movies { get; set; } = new();
}

File: src/frontend/src/models/actor.ts
export interface ActorResponse  {
  actor: ActorData;
  isFavorite: boolean;
}

export interface ActorData {
  id: string;
  firstName: string;
  lastName: string;
  biography: string;
  birthDate: string;
  photoUrl: string;
  photos: string[];
  movies: MovieCardResponse[];
}

export interface MovieCardResponse {
  id: string;
  title: string;
  year: number;
  posterUrl: string;
  userRating: number;
  ratingCount: number;
  characterName: string;
}